docs/DeveloperGuide.md

```diff
@@ -1,34 +1,422 @@
+   Total modules: 4
+   +--------------------------------------------------------------------------------------------------+
+   ```
+
+
+
+Below is a *sequence diagram* to illustrate the above example scenario.  
+
+```
+	// To do Sequence diagram here
+```
+
+
+
+#### 1.2. Add Category and Task Features
+
+Since add category and add task feature are implemented in a similar pattern, they can be explained together.
+
+The add category/task enable the user to add category/task/tag into Category/Task/Tag List.(One to one correspondence, which means the user can only add category to Category List, task to Task List).
+
+When the user first request to execute the **add** command*(assuming the command format given is valid)* to add a category/task by providing its name, the application will parse the input after `addm` /`addc` command word as the name for category/task. From here, there are **three** possible outcomes:
+
+1. The specified upper parent directory does not exist. -- No category/task will be added.
+2. The name of the specified category/task already exist. -- No category/task will be added.
+3. The specified upper parent directory exist and the name of the specified category/task does not exist. -- The specified category/task will be added.
+
+#### Feature Implementation
+
+This feature is facilitated by the `AddCategoryCommand` and `AddTaskCommand` class which add the corresponding category or task respectively.
+
+The above-stated two classes overrides the `execute()` method which extends from the abstract `AddCommand` class which extends from the abstract `Command` class. The `execute()` method's role is to execute the adding category/task operation and do necessary checks.
+
+The `AddCategoryCommand` and `AddTaskCommand` will first call the `getParentDirectory()` method to get the parent directory, then it will instantiate a `Category`/`Task` object and try to call `add` method in their respective parent directory class to add the new object, exception will be thrown according to the following rules:
+
+1. `ModuleNotFoundException` will be thrown if the specified module code does not exists in the `ArrayList` of Module when adding a new category or task.
+2. `CategoryNotFoundException` will be thrown if the specified category name does not exists in the `ArrayList` of Category in `CategoryManager` in `Module` when adding a new task.
+3. `IncorrectDirectoryLevelException` will be thrown if the user is performing the`addc`/`addt` command in the wrong directory without specifying their full parent directories.(For example, when user trying to execute `addc` in a `Root`/`Task` directory, or trying to execute `addt`  in a `Root`/`Module` directory)
+4. `DuplicateTaskException`/`DuplicateCategoryException` will be thrown if the name of the category/task already exist in the current directory.
+
+Below are the class-diagram for the involved classes:
+
+```
+to-do: add the class-diagram
```

Comment:
The description is quite clear but I still feel a bit hard to follow without a diagram.
Look forward to a detailed diagram!